src/bfuse16.rs

```diff
@@ -77,27 +77,42 @@ impl Filter<u64> for BinaryFuse16 {
     }
 }
 
+impl BinaryFuse16 {
+    /// Try to construct the filter from a key iterator. Can be used directly
+    /// if you don't have a contiguous array of u64 keys.
+    ///
+    /// Note: the iterator will be iterated over multiple times while building
+    /// the filter. If using a hash function to map the key, it may be cheaper
+    /// just to create a scratch array of hashed keys that you pass in.
+    fn try_from_iterator<T>(keys: T) -> Result<Self, &'static str>
+    where
+        T: ExactSizeIterator<Item = u64> + Clone,
+    {
+        bfuse_from_impl!(keys fingerprint u16, max iter 1_000)
+    }
+}
+
 impl TryFrom<&[u64]> for BinaryFuse16 {
     type Error = &'static str;
 
     fn try_from(keys: &[u64]) -> Result<Self, Self::Error> {
-        bfuse_from_impl!(keys fingerprint u16, max iter 1_000)
+        Self::try_from_iterator(keys.iter().map(|k| *k))
```

Comment:
Use `keys.iter().cloned()` instead

Reply:
`keys.iter().copied()` works too, I would prefer copy rather than clone explicitly

Reply:
`copied()` is more semantically correct 👍

Reply:
Nice. This also seems to improve performance. 
```
BinaryFuse16/from/500000                                                                           
                        time:   [9.3659 ms 9.3815 ms 9.3919 ms]
                        change: [-9.4493% -8.1909% -7.0114%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high severe

BinaryFuse16/contains/500000                                                                             
                        time:   [1.6878 ns 1.6911 ns 1.6956 ns]
                        change: [-2.1535% -1.6202% -1.0373%] (p = 0.00 < 0.05)
                        Performance has improved.
```